### PR TITLE
Referral promo fixes

### LINKED ISSUE #631: add DB constraint for promo id, referral code
we check uniqueness of promo referral codes at the app level, however we should also add a DB level constraint.
a unique multi-column index on promo_registrations, ["promo_id", "referral_code"] should do it

### LINKED ISSUE #642: Supply download links to promo assets in 'promo activated' emails
This was originally left out because we did not have the download links ready at the time of launch.

### LINKED ISSUE #645: Use https for underlying referral links in promo emails
The links are currently using the default`http`, we should change this to `https`.

cc @diracdeltas

### PR DESCRIPTION
**Pull request by nvonpentz**
Resolves #631 (add DB constraint for referral code, promo id)
Resolves #645 (use https for underlying links in promo emails)
Resolves #642 (supply download links to promo assets in promo emails) cc @jenn-rhim 

Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave-intl/publishers/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Tagged reviewers.

Test Plan:

Reviewer Checklist:

Tests
- [ ] Adequate test coverage exists to prevent regressions

Security:
- [ ] No raw SQL -- Always prefer ActiveRecord query helpers ([more info on StackOverflow](https://stackoverflow.com/questions/41410752/rails-5-sql-injection#41452695))
- [ ] XSS is mitigated -- Avoid `html_safe` and `raw`; escape untrusted content from users and 3rd party APIs ([Rails XSS guide](https://brakemanpro.com/2017/09/08/cross-site-scripting-in-rails) and [OWASP XSS Prevention Cheat Sheet](https://www.owasp.org/index.php/XSS_(Cross_Site_Scripting)_Prevention_Cheat_Sheet))

### CODE REVIEW

File: app/assets/stylesheets/mailer.scss [hunk 1]
```diff
@@ -194,6 +194,10 @@ body {
     .promo--ad-banner-youtube {
       float: left;
     }
+
+    .promo--download-link a{
```
Comment (ayumi) [inline app/assets/stylesheets/mailer.scss]: minor: should have a space after `a`

File: app/views/promo_mailer/new_channel_registered_2018q1.html.slim [hunk 2]
```diff
@@ -7,7 +7,7 @@ p = t("promo_mailer.promo_activated_2018q1_verified.body_one")
 
 a.promo--item-label.promo--item-label-link-name = @channel.details_type == "YoutubeChannelDetails" ? "#{@channel.publication_title.upcase} #{t("promo.shared.on_youtube")}" : "#{@channel.publication_title.upcase}"
 .promo--referral-link-container
-  .promo--referral-link = referral_url(@channel.promo_registration.referral_code)
+  .promo--referral-link = link_to(referral_url(@channel.promo_registration.referral_code), "https://" + referral_url(@channel.promo_registration.referral_code))
```
Comment (ayumi) [inline app/views/promo_mailer/new_channel_registered_2018q1.html.slim]: this is a little hacky.. should we not fix referral_url() to return the full "https://..." url?

Comment (nvonpentz) [reply app/views/promo_mailer/new_channel_registered_2018q1.html.slim]: this method is also used on the dashboard where we do not want to display.  I could write a new method or add an argument to this one if you prefer, but we do need to differentiate somehow.

Comment (ayumi) [reply app/views/promo_mailer/new_channel_registered_2018q1.html.slim]: i see. the clean way would be to have the existing display method to be `#human_referral_url`, and the complete one with HTTPS:// to be `#referral_url`. consider also a string is not actually a url if it doesn't have the protocol, which is the existing behavior.

Comment (nvonpentz) [reply app/views/promo_mailer/new_channel_registered_2018q1.html.slim]: cool, I will write a `#human_referral_url` and push the fix.  as far as strings not being urls, i'll have to think about that a bit more..

**Pull request merged**